src/pipe/modules/lenswarp/main.comp

```diff
@@ -0,0 +1,77 @@
+#version 460
+#extension GL_GOOGLE_include_directive    : enable
+
+#include "shared.glsl"
+#include "config.h"
+
+layout(local_size_x = DT_LOCAL_SIZE_X, local_size_y = DT_LOCAL_SIZE_Y, local_size_z = 1) in;
+
+layout(std140, set = 0, binding = 1) uniform params_t
+{
+  vec2 center;
+  float scale;
+  uint planes;
+} params;
+
+layout( // input f16 buffer rgb
+    set = 1, binding = 0
+) uniform sampler2D img_in;
+
+layout( // output f16 buffer rgb
+    set = 1, binding = 1
+) uniform writeonly image2D img_out;
+
+layout( // radius LUT
+  set = 1, binding = 2
+) uniform sampler2D radlut_in;
+
+void
+main()
+{
+  ivec2 ipos = ivec2(gl_GlobalInvocationID);
+  if(any(greaterThanEqual(ipos, imageSize(img_out)))) return;
+  if(params.planes == 0)
```

Comment:
for maximum performance in this case, there is [a bypass api call](https://github.com/hanatos/vkdt/blob/master/src/pipe/modules/resize/main.c#L24) like used in the resize module in the link. it will cpu side short-circuit the connectors so the shader will not even be called, avoiding a roundtrip from and to global memory. has the downside that we'll need to check if parameters changed  and in this case recreate the pipeline, but shaves off 0.1ms or so..